File: console/src/main/java/com/momoino/console/modules/auth/UserRepository.java
package com.momoino.console.modules.auth;

import java.util.Optional;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.lang.NonNull;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepository extends JpaRepository<UserEntity, UUID> {
    Optional<UserEntity> findByUsername(@NonNull String username);
}

File: backend/common/src/main/kotlin/com/momoino/common/auth/repository/UserRepository.kt
package com.momoino.common.auth.repository

import com.momoino.common.auth.entity.UserEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface UserRepository : JpaRepository<UserEntity, UUID> {
  fun findByUsername(username: String): UserEntity?
}
